services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -387,33 +403,41 @@ String getURLForFileHandle(FileHandle handle) {
 			ProxyStorageLocationSettings proxyStorage = (ProxyStorageLocationSettings) storage;
 			return ProxyUrlSignerUtils.generatePresignedUrl(proxyHandle, proxyStorage, new Date(System.currentTimeMillis() + PRESIGNED_URL_EXPIRE_TIME_MS));
 		} else if (handle instanceof S3FileHandle) {
-			S3FileHandle s3File = (S3FileHandle) handle;
-			// Create a pre-signed url
-			GeneratePresignedUrlRequest request = new GeneratePresignedUrlRequest(s3File.getBucketName(), s3File.getKey(), HttpMethod.GET);
-			request.setExpiration(new Date(System.currentTimeMillis() + PRESIGNED_URL_EXPIRE_TIME_MS));
-
-			ResponseHeaderOverrides responseHeaderOverrides = new ResponseHeaderOverrides();
-
-			String contentType = handle.getContentType();
-			if (StringUtils.isNotEmpty(contentType) && !NOT_SET.equals(contentType)) {
-				responseHeaderOverrides.setContentType(contentType);
-			}
-			String fileName = handle.getFileName();
-			if (StringUtils.isNotEmpty(fileName) && !NOT_SET.equals(fileName)) {
-				responseHeaderOverrides.setContentDisposition(ContentDispositionUtils.getContentDispositionValue(fileName));
-			}
-
-			request.setResponseHeaders(responseHeaderOverrides);
-			return s3Client.generatePresignedUrl(request).toExternalForm();
+			return getUrlForS3FileHandle((S3FileHandle) handle);
+		} else if (handle instanceof GoogleCloudFileHandle) {
+			return getUrlForGoogleCloudFileHandle((GoogleCloudFileHandle) handle);
 		} else if (handle instanceof ExternalObjectStoreFileHandle){
 			ExternalObjectStoreFileHandle fileHandle = (ExternalObjectStoreFileHandle) handle;
 			return StringUtils.join(new String[]{fileHandle.getEndpointUrl(), fileHandle.getBucket(), fileHandle.getFileKey()} , '/');
 		} else {
-			throw new IllegalArgumentException("Unknown FileHandle class: "
-					+ handle.getClass().getName());
+			throw new IllegalArgumentException("Unknown FileHandle class: " + handle.getClass().getName());
 		}
 	}
 
+	private String getUrlForS3FileHandle(S3FileHandle handle) {
+		// Create a pre-signed url
+		GeneratePresignedUrlRequest request = new GeneratePresignedUrlRequest(handle.getBucketName(), handle.getKey(), HttpMethod.GET);
+		request.setExpiration(new Date(System.currentTimeMillis() + PRESIGNED_URL_EXPIRE_TIME_MS));
+
+		ResponseHeaderOverrides responseHeaderOverrides = new ResponseHeaderOverrides();
+
+		String contentType = handle.getContentType();
```

Comment:
override content disposition logic is needed for google as well.  file rename will not work as expected without it.